core-skills/c-sharp-and-dotnet/README.md

```diff
@@ -0,0 +1,78 @@
+4. Understands basic concepts of the C# language and ecosystem
+
+### Goat
+1. Understands some deeper object oriented programming concepts
+2. Able to demonstrate common core patterns implemented using C# language features
+3. Understands how to build API functionality using common frameworks (MVC, EF)
+4. Understands how to use the dotnet command line utilities to perform common tasks
+5. Can effectively debug using the interactive debugger (IDE)
+
+### Eagle
+1. Demonstrate an understanding of functional programming in C#
+2. Understanding of the .NET virtual machine
+3. Understanding the impact of choices on performance
+4. Demonstrate ability to combine understanding of the above into a working API in C# with .NET
+
+## Resources
+Coming soon
+
+## Mark Scheme
+### Mantis Shrimp
+
+- Can explain the difference between .NET, .NET Core and Framework
+- Can explain the difference between C#, Visual Basic and F#
+- Can explain what assemblies and artefacts are
+- Can explain the concept of classes with examples of usage
+- Can explain the concept of interfaces with examples of usage
+- Can explain the concept of inheritance with examples of usage
+
+- Can create a new project in an IDE
+- Can install and manage external project dependancies
+- Can build and run the project
+- Can manage internal project dependancies
+- Can create a test project and a suite of tests that test some functionality
+- Demonstrates use of classes, interfaces and inheritance
+- Demonstrates the use of scope/access modifiers
+
+Expected deliverable - simple application with a test that satisfies the above
+
+### Goat
+
+- Can explain the concept of generics with examples of usage
+- Can explain the concept of abstraction with examples of usage
+- Can explain the concept of polymorphism with examples of usage
+
+- Can demonstrate understanding and implementation of exception handling
+- Can demonstrate understanding and implementation of asynchronous programming
+- Can demonstrate understanding and implementation of LINQ query syntax and method syntax
+- Can demonstrate understanding and implementation of data persistence
+- Can demonstrate understanding and implementation of SOLID principles
+- Can demonstrate effectively debugging using an interactive debugging tool
+
+Expected deliverable - HTTP API that can take a request and persist some data from that request demonstrating the above
+
+### Eagle
+Disclaimer - The C# skill is not a core skill, it is advanced learning. If the goal is to know enough to join a delivery, this is satisfied having achieved Goat.
+Eagle is a space to demonstrate a deeper understanding of OO programming and implementing clean, maintainable code in C#.
+
+- Can explain at a high level how the .NET virtual machine works
+ - Can explain memory management and garbage collection
+ - Can explain intermediate language and compilation
+- Can explain common performance detriments in C# code and some potential ways to write performant C#
```

Comment:
Does `clean, maintainable code` require performant code?

Reply:
No. In fact, performance is often a premature optimisation. This is why performant code is in the third and final tier. We felt that achieving Eagle requires some understanding of both _clean_ and _performant_, though the two may not necessarily be intertwined.